python/sparkdl/graph/builder.py

```diff
@@ -0,0 +1,251 @@
+        """
+        Import a GraphFunction object into the current session
+
+        :param gfn: GraphFunction, an object representing a TensorFlow graph and its inputs and outputs
+        :param input_map: dict, mapping from input names to existing graph elements
+        :param name: str, the scope for all the variables in the GraphFunction's elements
+        :param gdef_kargs: other keyword elements for TensorFlow's `import_graph_def`
+        """
+        try:
+            del gdef_kargs["return_elements"]
+        except KeyError:
+            pass
+        if input_map is not None:
+            assert set(input_map.keys()) <= set(gfn.input_names), \
+                "cannot locate provided input elements in the graph"
+
+        input_names = gfn.input_names
+        output_names = gfn.output_names
+        if name is not None:
+            name = name.strip()
+            if len(name) > 0:
+                output_names = [
+                    name + '/' + op_name for op_name in gfn.output_names]
+                input_names = [
+                    name + '/' + op_name for op_name in gfn.input_names]
+
+        # When importing, provide the original output op names
+        tf.import_graph_def(gfn.graph_def,
+                            input_map=input_map,
+                            return_elements=gfn.output_names,
+                            name=name,
+                            **gdef_kargs)
+        feeds = [tfx.get_tensor(self.graph, name) for name in input_names]
+        fetches = [tfx.get_tensor(self.graph, name) for name in output_names]
+        return (feeds, fetches)
+
+
+class GraphFunction(object):
+    """
+    Represent a TensorFlow graph with its GraphDef, input and output operation names.
+
+    :param graph_def: GraphDef, a static ProtocolBuffer object holding informations of a TensorFlow graph
+    :param input_names: names to the input graph elements (must be of Placeholder type)
+    :param output_names: names to the output graph elements
+    """
+
+    def __init__(self, graph_def, input_names, output_names):
+        """
+        :param graph_def: GraphDef object
+        :param input_names: list of input (operation) names (must be typed `Placeholder`)
+        :param output_names: list of output (operation) names
+        """
+        self.graph_def = graph_def
+        self.input_names = input_names
+        self.output_names = output_names
+
+    def dump(self, fpath):
+        """
+        Store the GraphFunction to a file
+
+        :param fpath: str or path, path to the serialized GraphFunction
+        """
+        serialized = {"graph_def_bytes": self.graph_def.SerializeToString(),
+                      "inputs": self.input_names,
+                      "outputs": self.output_names}
+        assert isinstance(fpath, six.string_types)
+        if not fpath.endswith("gfn"):
+            fpath += ".gfn"
+        with open(fpath, 'wb') as fout:
+            pickle.dump(serialized, fout)
```

Comment:
Why aren't we using json here? Isn't serialized made up of strings and lists? json is much better for portability than pickle.